cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator.cpp

```diff
@@ -0,0 +1,472 @@
+            break;
+    }
+
+    return ib;
+}
+/* END TrajectoryProfileBase *******************************************************************************************/
+
+/* BEGIN TrajectoryProfileRamp ********************************************************************************************/
+inline void TrajectoryProfileRamp::getProfileTimings(double Se_max, double accl, double vel)
+{
+        int steps_te, steps_tv, steps_tb = 0;
+        double tv, tb, te = 0.0;
+
+        // Calculate the Ramp Profile Parameters
+        if (vel > sqrt(std::fabs(Se_max) * accl))
+        {
+            vel = sqrt(std::fabs(Se_max) * accl);
+        }
+        tb = vel / accl;
+        te = (std::fabs(Se_max) / vel) + tb;
+        tv = te - tb;
+
+        // Interpolationsteps for every timesequence
+        steps_tb = trunc(tb      / params_.profile.t_ipo);
+        steps_tv = trunc((tv-tb) / params_.profile.t_ipo);
+        steps_te = trunc((te-tv) / params_.profile.t_ipo);
+
+        // Reconfigure timings wtih t_ipo_
+        pt_.tb = steps_tb * params_.profile.t_ipo;
+        pt_.tv = (steps_tb + steps_tv) * params_.profile.t_ipo;
+        pt_.te = (steps_tb + steps_tv + steps_te) * params_.profile.t_ipo;
+        pt_.steps_tb = steps_tb;
+        pt_.steps_tv = steps_tv;
+        pt_.steps_te = steps_te;
+}
+
+inline bool TrajectoryProfileRamp::generatePath(cob_cartesian_controller::PathArray &pa)
+{
+    std::vector<double> array;
+    double accl = params_.profile.accl;
+
+    double direction = pa.getSe()/std::fabs(pa.getSe());
+
+    if(pa.getCalcTe())
+    {
+        ROS_INFO_STREAM("Longest Se.. Vel_max: " << params_.profile.vel << "   accl_max: " << params_.profile.accl << "   tb: " << pt_.tb << "   tv: " << pt_.tv << "   te: " << pt_.te);
+
+        int i = 1;
+        // Calculate the ramp profile path
+        // 0 <= t <= tb
+        for(; i <= pt_.steps_tb ; i++)
+        {
+            array.push_back(pa.getStartValue() + direction * (0.5*params_.profile.accl*pow((params_.profile.t_ipo*i),2)));
+        }
+
+        // tb <= t <= tv
+        for(; i <= (pt_.steps_tb + pt_.steps_tv) ; i++)
+        {
+            array.push_back(pa.getStartValue() + direction * (params_.profile.vel*(params_.profile.t_ipo*i) - 0.5 * pow(params_.profile.vel,2)/accl)  );
+        }
+
+        // tv <= t <= te
+        for(; i <= (pt_.steps_tv + pt_.steps_tb + pt_.steps_te + 1) ; i++)
+        {
+            array.push_back(pa.getStartValue() + direction *(params_.profile.vel * (pt_.tv) - 0.5*accl* pow(pt_.te-(i*params_.profile.t_ipo),2)));
+        }
+    }
+    else
+    {
+        if(std::fabs(pa.getSe()) > 0.001)
+        {
+            double vel_max;
+            double accl_max = params_.profile.accl;
+            double tb = pt_.tb;
+            double tv = pt_.tv;
+            double te = pt_.te;
+
+            double steps_tb, steps_te, steps_tv;
+            // Calculate the Profile Timings
+            // Reconfigure accl_max and Velmax
+
```

Comment:
the following pretty much looks like what is done in computeProfileTimings....I bet this can be unified somehow!